sympy/vector/coordsysrect.py

```diff
@@ -98,6 +98,53 @@ def __new__(cls, name, location=None, rotation_matrix=None,
             location = Vector.zero
             origin = Point(name + '.origin')
 
+        if transformation is None:
+            if parent is None:
+                if rotation_matrix is not None:
+                    rotation_equations = \
+                       lambda x, y, z: CoordSys3D._rotation_trans_equations(parent_orient, lambda x, y, z: (x, y, z))
+                else:
+                    rotation_equations = lambda x, y, z: (x, y, z)
+
+                # Information about location without parent doesn't have sense.
+                # See lines 80:99. Location is somehow treated only when parent is put.
+                # We are not able to create independent vector, so we can't assign
+                # anything to location.
+                translation_equations = lambda x, y, z: (x, y, z)
+
+                from sympy import symbols
+                x, y, z = symbols('x y z')
+                transformation_equations = rotation_equations(*translation_equations(x, y, z))
+                if not CoordSys3D._check_orthogonality(transformation_equations, (x, y, z)):
+                    raise ValueError("The transformation equation does not "
+                                     "create orthogonal coordinate system")
+
+            else:
+                if rotation_matrix is not None:
+                    rotation_equations = \
+                        lambda x, y, z: parent._rotation_trans_equations(parent_orient, (x, y, z))
+                else:
+                    rotation_equations = lambda x, y, z: (x, y, z)
+
+                if location is not None:
+                    translation_equations = \
+                        lambda x, y, z: parent._translation_trans_equations(parent, origin, (x, y, z))
+                else:
+                    translation_equations = lambda x, y, z: (x, y, z)
+
+                transformation_equations = rotation_equations(*translation_equations(*parent.base_scalars()))
+                if not CoordSys3D._check_orthogonality(transformation_equations, parent.base_scalars()):
+                    raise ValueError("The transformation equation does not "
+                                     "create orthogonal coordinate system")
+        else:
+            if not (location and rotation_matrix) is None:
+                raise ValueError
+
+            if parent is None:
+                transformation_equations = None
```

Comment:
I didn't implement this part yet.
I just put `None` to create some structure for case when `transformation` is not `None`.